### PR TITLE
Use https URLs to refer to rust-lang.org where appropriate.

### PR DESCRIPTION
**Pull request by eefriedman**
Also fixes a few outdated links.

### CODE REVIEW

File: src/doc/style/features/functions-and-methods/input.md [hunk 1]
```diff
@@ -156,8 +156,7 @@ over
 fn foo(a: u8) { ... }
 ```
 
-Note that
-[`ascii::Ascii`](http://static.rust-lang.org/doc/master/std/ascii/struct.Ascii.html)
+Note that `ascii::Ascii`
```
Comment (steveklabnik) [inline src/doc/style/features/functions-and-methods/input.md]: why lose the link here?

File: src/test/run-pass/cleanup-arm-conditional.rs [hunk 2]
```diff
@@ -8,16 +8,6 @@
...
 
-// copyright 2014 the rust project developers. see the copyright
-// file at the top-level directory of this distribution and at
-// http://rust-lang.org/copyright.
-//
-// licensed under the apache license, version 2.0 <license-apache or
-// http://www.apache.org/licenses/license-2.0> or the mit license
-// <license-mit or http://opensource.org/licenses/mit>, at your
-// option. this file may not be copied, modified, or distributed
-// except according to those terms.
```
Comment (steveklabnik) [inline src/test/run-pass/cleanup-arm-conditional.rs]: this probably should not be going away

File: src/test/run-pass/cleanup-shortcircuit.rs [hunk 3]
```diff
@@ -8,16 +8,6 @@
...
-// copyright 2014 the rust project developers. see the copyright
-// file at the top-level directory of this distribution and at
-// http://rust-lang.org/copyright.
-//
-// licensed under the apache license, version 2.0 <license-apache or
-// http://www.apache.org/licenses/license-2.0> or the mit license
-// <license-mit or http://opensource.org/licenses/mit>, at your
-// option. this file may not be copied, modified, or distributed
-// except according to those terms.
-
```
Comment (steveklabnik) [inline src/test/run-pass/cleanup-shortcircuit.rs]: same

File: src/test/run-pass/cleanup-arm-conditional.rs (hunk 2 above)
Comment (eefriedman) [inline src/test/run-pass/cleanup-arm-conditional.rs]: The full copyright statement is still there; it just got duplicated somehow.

File: src/doc/style/features/functions-and-methods/input.md (hunk 1 above)
Comment (eefriedman) [inline src/doc/style/features/functions-and-methods/input.md]: `ascii::Ascii` no longer exists in the standard library.

Comment (steveklabnik) [inline src/doc/style/features/functions-and-methods/input.md]: Then we probably should remove this bit entirely?

Comment (eefriedman) [inline src/doc/style/features/functions-and-methods/input.md]: I can rewrite it, I guess.

Comment (eefriedman) [inline src/doc/style/features/functions-and-methods/input.md]: Replaced `ascii::Ascii` with a different example.

**Comment by alexcrichton** [comment]
@bors: r+ 7d3c4bd72a8e12e926592ef5592d188c76ac6b7a

Thanks!

**Pull request merged**